x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx

```diff
@@ -118,16 +125,49 @@ const sortTableData =
  * Build a table columns and items starting from the snapshot nodes.
  */
 export const useHostsTable = () => {
+  const [selectedItems, setSelectedItems] = useState<HostNodeRow[]>([]);
   const { hostNodes } = useHostsViewContext();
   const { searchCriteria } = useUnifiedSearchContext();
   const [{ pagination, sorting }, setProperties] = useHostsTableUrlState();
   const {
-    services: { telemetry },
+    services: {
+      telemetry,
+      data: {
+        query: { filterManager: filterManagerService },
+      },
+    },
   } = useKibanaContextForPlugin();
+  const { dataView } = useMetricsDataViewContext();
+
   const [hostFlyoutState, setHostFlyoutState] = useHostFlyoutUrlState();
   const popoverContainerRef = React.createRef<HTMLDivElement>();
 
   const closeFlyout = useCallback(() => setHostFlyoutState(null), [setHostFlyoutState]);
+
+  const onSelectionChange = (newSelectedItems: HostNodeRow[]) => {
+    setSelectedItems(newSelectedItems);
+  };
+
+  const hostsTableRef = useRef<EuiBasicTable | null>(null);
```

Comment:
nit: For readability, could we move this line next to the `popoverContainerRef`?